### PR TITLE
Improve z axis test

### PR DESCRIPTION
**Pull request by qlthd**
## Description

Added test verifying tooltip content, dots size enriched with z axis.
## Related Issue

[#4591](https://github.com/recharts/recharts/issues/4591
)

## Motivation and Context

Lack of test for ZAxis component
## How Has This Been Tested?

Test

## Types of changes

Test

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

**Comment by ckifer** [comment]
@qlthd looks like maybe you're making this PR starting from the `master` branch. Can you start from `3.x`?

**Review by ckifer — APPROVED** [review]
Good start to me! Probably need to do some assertions that the value of the ZIndex changes the size of the scatter dot too

Will wait for another review before merging

**Review by PavelVanecek — APPROVED** [review]
Nice! Thanks for adding this.

If you're looking for what to do for the next PR, there's a few more scenarios we need covered too:

- Tooltip with `defaultIndex`
- Tooltip with `active`
- Tooltip with `trigger='click'`
- ZAxis + XAxis, ZAxis + YAxis, ZAxis alone, without other axes
- `ComposedChart` as the parent element instead of `ScatterChart`
- Multiple ZAxis
- And combinations of above

**Comment by PavelVanecek** [comment]
I will go ahead and merge this, I am changing the axis code now so I will appreciate having test coverage. We can iterate with new expectations and test scenarios in separate PR.

**Pull request merged**
